buyer-api/src/contractEvents/dataOrderSubscriber.js

```diff
@@ -0,0 +1,26 @@
+import apicache from 'apicache';
+import { getAccount } from '../services/signingService';
+import { fetchDataOrder } from '../utils/blockchain';
+import { dataOrders } from '../utils/stores';
+
+export const dataOrderCacheSubscriber = {
+  name: 'CacheUpdater',
+  events: [
+    'DataOrderCreated',
+    'DataOrderClosed',
+  ],
+  async callback({ returnValues: { orderId: id, owner } }) {
+    const { address } = await getAccount();
+    if (address === owner) {
+      const [storedOrder, chainOrder] = await Promise.all([
+        dataOrders.fetch(id),
+        fetchDataOrder(id),
+      ]);
```

Comment:
Remember that the `orderId` param that comes in the event does not identify a DataOrder in the local db yet, but it identifies the order in the DataExchange. I'd change this do the `fetchDataOrder(id)` first, and then update the local version of the order with the new `id` field.
Also, the order `status` is another field thats needs to be updated to `created`